app/models/supplejack_api/enrichable.rb

```diff
@@ -3,40 +3,8 @@ module SupplejackApi
   module Enrichable
     extend ActiveSupport::Concern
 
-    included do
-      include Mongoid::Document
-      include Mongoid::Attributes::Dynamic
-
-      store_in client: 'api'
-
-      embeds_many :fragments, cascade_callbacks: true, class_name: 'SupplejackApi::ApiRecord::RecordFragment'
-      delegate :title, :shelf_location, :relation, to: :primary
-    end
-
     def primary
-      fragments.where(priority: 0).first
-    end
-
-    def parent_tap_id
-      extract_tap_id(:is_part_of) || extract_tap_id(:relation)
-    end
-
-    def tap_id
-      extract_tap_id(:dc_identifier)
-    end
-
-    def authority_taps(name)
-      primary.authorities.map { |authority| authority.authority_id if authority.name == name.to_s }.compact
-    end
-
-    def authorities
-      authorities = {}
-      sorted_fragments.each do |fragment|
-        fragment.authorities.each do |authority|
-          authorities["#{authority.authority_id}-#{authority.name}"] ||= authority
-        end
-      end
-      authorities.values
+      fragments.select { |a| a.priority == 0 }.first
```

Comment:
You can use find instead of select.first :)

Reply:
We avoided using find because the old code was written with in a way that only takes the first of possibly many results. find will return many results potentially, and we would need to call.first on it too.